src/routes/suggestions/find.ts

```diff
@@ -0,0 +1,70 @@
+import Joi from 'joi';
+import { ControllerGeneratorOptionsWithClient } from '../../comms/controller';
+
+const suggestionTypes = [
+  'clientReportType.name',
+  'commandType.name',
+  'deviceType.name',
+  'edge.name',
+  'grid.name',
+  'issue.title',
+  'pinGroup.name',
+  'pin.name',
+  'quantity.name',
+  'quantity.unit',
+  'user.name',
+] as const;
+
+type SuggestionType = (typeof suggestionTypes)[number];
+
+interface Request {
+  query: {
+    search: string;
+    type: SuggestionType;
+    rowsPerPage: number;
+    offset?: string;
+  };
+}
+
+type EffectiveRequest = {
+  query: Required<Request['query']>;
+}
```

Comment:
I think offset is still optional in the EffectiveRequest, isn't it?

Reply:
updated